lib/rubocop/cop/style/zero_length_predicate.rb

```diff
@@ -0,0 +1,52 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for receiver.length == 0 predicates and the
+      # negated versions receiver.length > 0 and receiver.length != 0.
+      # These can be replaced with receiver.empty? and
+      # !receiver.empty? respectively.
+      #
+      # @example
+      #
+      #   @bad
+      #   [1, 2, 3].length == 0
+      #   0 == "foobar".length
+      #   hash.size > 0
+      #
+      #   @good
+      #   [1, 2, 3].empty?
+      #   "foobar".empty?
+      #   !hash.empty?
+      class ZeroLengthPredicate < Cop
+        ZERO_MSG = 'Use `empty?` instead of `%s == 0`.'.freeze
+        NONZERO_MSG = 'Use `!empty?` instead of `%s %s %s`.'.freeze
+
+        def on_send(node)
+          zero_length_predicate = zero_length_predicate(node)
+          if zero_length_predicate
```

Comment:
I'd add a few blank lines around those conditionals to increase the readability a bit.